### PR TITLE
fix(menu): menu-title-content style

### LINKED ISSUE #51361: [Menu] 导航菜单中，菜单项标题 label 中，Flex 布局设置的属性失效
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-43xydr)

### Steps to reproduce

1. 打开 CodeSandbox 链接。
2. 在侧边栏中查看菜单项的布局。
3. 注意菜单项 `label` 中的 `justifyContent: 'space-between'` 属性无法正常工作，导致右侧的下拉按钮与左侧的文本没有正确对齐。

若想查看预期效果，请参考使用 antd 版本 5.20.6 的 [CodeSandbox 链接](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-jzyy3m)。

### What is expected?

`justifyContent: 'space-between'` 应该使 `label` 中的元素（文本和下拉按钮）在左右两边对齐。

### What is actually happening?

`justifyContent: 'space-between'` 属性失效，导致文本和下拉按钮没有正确对齐，按钮紧靠文本显示。

| Environment | Info |
| --- | --- |
| antd | 5.21.0 |
| React | 18.3.1 |
| System | macOS 15.0.1 |
| Browser |  Google Chrome 129.0.6668.101 (arm64) |

---
使用 Ant Design 5.20.6 版本时，布局是正常的。升级到 5.21.x 后，问题开始出现，`label` 中的 Flex 布局属性没有正确应用。

### LINKED ISSUE #51469: [Menu] extra node is in the wrong position
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-xz8yzm)

官网效果

image

### Steps to reproduce

脱离官网

### What is expected?

extra 在右侧

### What is actually happening?

extra 没在右侧

| Environment | Info |
| --- | --- |
| antd | 5.21.6 |
| React | 18.3.1 |
| System | Windows |
| Browser | Chrome  |

### LINKED ISSUE #51526: Menu组件文本超出宽度被截断，没有省略
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-bdqztc-ktc7eb?file=demo.tsx)

### Steps to reproduce

打开重现链接，查看Menu侧边栏

### What is expected?

期望文本不被截断，可以出现省略号，鼠标hover上去出现完整的文案

### What is actually happening?

文本超出宽度 被截断了，有的还出现了多余的省略号

| Environment | Info |
| --- | --- |
| antd | 5.21.6 |
| React | 18.3.4 |
| System | mac pro |
| Browser | 谷歌128.0.6613.120（正式版本） (arm64) |

---
看了一下可能是display：inline这个样式影响的，去掉这个样式可以正常展示

### PR DESCRIPTION
**Pull request by coding-ice**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
- fix https://github.com/ant-design/ant-design/issues/51361
- fix https://github.com/ant-design/ant-design/issues/51469
- fix #51526

### 💡 Background and Solution
在 5.21.0 中，上了 `extra` 属性，变更 `ant-menu-title-content` 为 `inline-flex`
- 导致这个元素的宽度从原来的 100% 变为子元素决定，引发了样式的 `break`

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 1. Fix css break caused by menu-title-content flex layout <br /> 2. Fixed an issue where the color of theme: 'dark' was not obvious          |
| 🇨🇳 Chinese | 1. 修复 menu-title-content：flex 布局 引发的 css break <br/> 2. 修复 theme: 'dark' 的颜色不明显问题 |

### CODE REVIEW

File: components/dropdown/style/index.ts [hunk 1]
```diff
@@ -242,10 +242,13 @@ const genBaseStyle: GenerateStyle<DropdownToken> = (token) => {
...
 
           [`${menuCls}-title-content`]: {
-            display: 'flex',
-            alignItems: 'center',
             flex: 'auto',
 
+            '&-extra': {
+              display: 'inline-flex',
+              alignItems: 'center',
+            },
```
Comment (zombieJ) [inline components/dropdown/style/index.ts:250]: demo 里加一条吧，这样好 image diff 看出来~

**Comment by li-jia-nan** [comment]
这个现在什么进度啦？

**Comment by coding-ice** [comment]
@zombieJ 麻烦再看下～

File: components/dropdown/style/index.ts [hunk 2]
```diff
@@ -242,10 +242,14 @@ const genBaseStyle: GenerateStyle<DropdownToken> = (token) => {
           },
 
           [`${menuCls}-title-content`]: {
-            display: 'flex',
-            alignItems: 'center',
             flex: 'auto',
 
+            '&-extra': {
```
Comment (zombieJ) [inline components/dropdown/style/index.ts:247]: `-extra` 特指 extra 的 dom，不会做 layout。这里如果要加的话应该叫 `-with-extra`

**Review by zombieJ — APPROVED** [review]

**Pull request merged**